usecases/usecases.go

```diff
@@ -0,0 +1,263 @@
+package usecases
+
+import (
+	"fmt"
+
+	"game-tracker/domain"
+)
+
+type UserRepository interface {
+	Store(user User) error
+	Remove(user User) error
+	FindById(id int) (User, error)
+	Count() (int, error)
+	UserExisted(userName string) (bool, error)
+	StoreInfo(user User, info string) error
+	LoadInfo(user User) (string, error)
+}
+
+type LibraryRepository interface {
+	Store(library Library) error
+	Remove(library Library) error
+	FindById(id int) (Library, error)
+	Count() (int, error)
+}
+
+type GameRepository interface {
+	Store(game Game) error
+	Remove(game Game) error
+	FindById(id int) (Game, error)
+}
+
+type User struct {
+	Id           int
+	Name         string
+	Player       domain.Player //This user (account) was created by some player
+	PersonalInfo string
+}
+
+type Library struct {
+	Id    int
+	User  User //This library belongs to some user
+	Games []Game
+}
+
+type Game struct {
+	Id        int
+	LibraryId int
+	Name      string
+	Producer  string
+	Value     []uint8
+}
+
+type LoggerRepository interface {
+	Log(message string) error
+}
+
+type ProfileInteractor struct {
+	UserRepository    UserRepository
+	LibraryRepository LibraryRepository
+	GameRepository    GameRepository
+	Loggr             LoggerRepository
+}
+
+func (interactor *ProfileInteractor) ShowLibrary(userId, libraryId int) (string, []Game, error) {
+	var games []Game
+	user, err := interactor.UserRepository.FindById(userId)
+	if err != nil {
+		err = fmt.Errorf(fmt.Sprintf("User #%d does not exist", userId))
+		return "", nil, err
+	}
+	info, err := interactor.UserRepository.LoadInfo(user)
+	if err != nil {
+		err = fmt.Errorf("Error loading user personal information")
+		return "", nil, err
+	}
+	library, err := interactor.LibraryRepository.FindById(libraryId)
+	if err != nil {
+		err = fmt.Errorf(fmt.Sprintf("Library #%d of user #%d does not exist", libraryId, userId))
+		return "", nil, err
+	}
+
+	if user.Id != libraryId {
+		message := "User #%d is not allowed to see games in library #%d of user #%d"
+		err := fmt.Errorf(message, user.Id, libraryId, library.User.Id)
+		games = make([]Game, 0)
+		return info, games, err
+	} else {
+		games = make([]Game, len(library.Games))
+		for i, game := range library.Games {
+			games[i] = Game{Id: game.Id, Name: game.Name, Producer: game.Producer, Value: game.Value}
+		}
+		return info, games, nil
+	}
+}
+
+func (interactor *ProfileInteractor) AddUser(player domain.Player, userName string) error {
+	userCount, err := interactor.UserRepository.Count()
+	if err != nil {
+		return err
+	}
+	// Application rule: usernames cannot repeat
+	existed, err := interactor.UserRepository.UserExisted(userName)
+	if err != nil {
+		return err
+	}
+	if existed {
+		err := fmt.Errorf("Username '%s' is taken", userName)
+		// interactor.Logger.Log(err.Error())
+		return err
+	}
+	user := User{Id: userCount + 1, Name: userName, Player: player, PersonalInfo: ""}
```

Comment:
`Id` do nothing in `UserRepository.Store(user)` , so remove `userCount + 1`.
And if in your database, column `id` has been set `SERIAL` then don't need to count it up.
http://www.tutorialspoint.com/postgresql/postgresql_using_autoincrement.htm